csds-client/client/v3/client.go

```diff
@@ -0,0 +1,390 @@
+// Package client/v3 implements the client interface for v3 transport api version in specific
+package client
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"envoy-tools/csds-client/client"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"time"
+
+	csdspb_v3 "github.com/envoyproxy/go-control-plane/envoy/service/status/v3"
+	envoy_type_matcher_v3 "github.com/envoyproxy/go-control-plane/envoy/type/matcher/v3"
+	"github.com/ghodss/yaml"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+// ClientV3 implements the interface Client
+type ClientV3 struct {
+	clientConn *grpc.ClientConn
+	csdsClient csdspb_v3.ClientStatusDiscoveryServiceClient
+
+	nodeMatcher []*envoy_type_matcher_v3.NodeMatcher
+	metadata    metadata.MD
+	opts        client.ClientOptions
+}
+
+// parseNodeMatcher parses the csds request yaml from -request_file and -request_yaml to nodematcher
+// if -request_file and -request_yaml are both set, the values in this yaml string will override and
+// merge with the request loaded from -request_file
+func (c *ClientV3) parseNodeMatcher() error {
+	if c.opts.RequestFile == "" && c.opts.RequestYaml == "" {
+		return errors.New("missing request yaml")
+	}
+
+	var nodematchers []*envoy_type_matcher_v3.NodeMatcher
+	if err := parseYaml(c.opts.RequestFile, c.opts.RequestYaml, &nodematchers); err != nil {
+		return err
+	}
+
+	c.nodeMatcher = nodematchers
+
+	// check if required fields exist in nodematcher
+	switch c.opts.Platform {
+	case "gcp":
+		keys := []string{"TRAFFICDIRECTOR_GCP_PROJECT_NUMBER", "TRAFFICDIRECTOR_NETWORK_NAME"}
```

Comment:
Should these be consts defined at the package level?

Reply:
Defined them as `gcpProjectNumberKey` and `gcpNetworkNameKey`.